cypress/integration/gallery_spec.js

```diff
@@ -0,0 +1,103 @@
+describe('Gallery', () => {
+  beforeEach(() => {
+    //Add additional intercept here for photos
+    cy.intercept("GET", "https://elegy-backend.herokuapp.com/api/v1/users/2", {
+        ok: true,
+        status: 200, 
+        url: "https://elegy-backend.herokuapp.com/api/v1/users/2",
+        fixture: 'user'
+    })
+    cy.visit("http://localhost:3000/")
+    cy.get(".gallery-pane")
+      .click()
+  })
+
+  it('Should be able to visit the gallery page by typing in the path in the url', () => {
+    cy.url().should("include", "http://localhost:3000/gallery")
+  })
+
+  it('Should tell user which page they are visiting currently', () => {
+    cy.get(".gal-head")
+      .contains("Gallery")
+  })
+
+  it('Should direct the user to add a profile photo if there is none', () => {
+    cy.get(".gal-rule")
+    .should("contain", "Please add a Profile Photo")
+  })
+
+  it('Should display the user profile photo in the center pane if there is one', () => {
+    cy.get("h1")  
+      .click()
+    cy.fixture("william.png")
+      .then(file => Cypress.Blob.base64StringToBlob(file))
+      .then((fileContent) => {
+          cy.get(".sun > img")
+          .click()
+          .get(".photo-upload")
+          .attachFile({
+              fileContent: fileContent,
+              fileName: "william.png",
+              mimeType: "image/png"
+          })
+          .get(".photo-edit-button")
+          .click()
+          cy.get(".sun > img")
+          .should("have.attr", "src").should("include", "blob:http://localhost:3000")
+        cy.get(".sun > img")
+          .click()
+          .get(".user-image")
+          .should("have.attr", "src").should("include", "blob:http://localhost:3000")
+      })
+    cy.get("h1")  
+      .click()
+    cy.get(".gallery-pane")
+      .click()
+    cy.get(".gal-prof")
+      .should("have.attr", "src").should("include", "blob:http://localhost:3000")
+  })
+
+  it('Should have a button for the user to add a photo', () => {
+      cy.get(".add-gal-photo")
+      .should("be.visible")
+  })
+
+  it('Should redirect user to add photo view', () => {
+      cy.get(".add-gal-photo")
+      .click()
+      cy.url().should("include", "http://localhost:3000/add-photo/gallery")
+  })
+
+  it("Should allow user to add photos to their gallery", () => {
+    cy.fixture("william.png")
+      .then(file => Cypress.Blob.base64StringToBlob(file))
+      .then((fileContent) => {
+        cy.get(".add-gal-photo")
+        .click()
+          .get(".photo-upload")
+          .attachFile({
+              fileContent: fileContent,
+              fileName: "william.png",
+              mimeType: "image/png"
+          })
+          .get(".photo-edit-button")
+          .click()
+        cy.get(".g-14")
+          .children("img")
+          .should("be.visible")
+          .should("have.attr", "src").should("include", "blob:http://localhost:3000")
```

Comment:
So the only thing with this test is that I'm just making sure that the first part of the source is correct because the rest is changing. You would not be able to test if you had uploaded different photos. just FYI